spec.bs

```diff
@@ -875,19 +875,15 @@ To <dfn>validate and convert auction ad config</dfn> given an {{AuctionAdConfig}
       1. Set |auctionConfig|'s [=auction config/seller signals=] to the result of
         [=serializing a JavaScript value to a JSON string=], given |result|.
     * To handle an error, set |auctionConfig|'s [=auction config/seller signals=] to failure.
-1. If |config|["{{AuctionAdConfig/directFromSellerSignals}}"] [=map/exists=]:
-  1. TODO: The receiving end of this isn't specified yet, so there is no place to put the computed value.
-  1. [=Handle an input promise in configuration=] given |auctionConfig| and |config|["{{AuctionAdConfig/directFromSellerSignals}}"]:
+1. If |config|["{{AuctionAdConfig/directFromSellerSignalsHeaderAdSlot}}"] [=map/exists=]:
+  1. [=Handle an input promise in configuration=] given |auctionConfig| and |config|["{{AuctionAdConfig/directFromSellerSignalsHeaderAdSlot}}"]:
     * To parse the value |result|:
-      1. Let |directFromSellerSignalsPrefix| be the result of running the [=URL parser=] on |result|.
-      1. [=exception/Throw=] a {{TypeError}} if any of the following conditions hold:
-        * |directFromSellerSignalsPrefix| is failure;
-        * |directFromSellerSignalsPrefix| is not [=same origin=] with |auctionConfig|'s
-          [=auction config/seller=];
-        * |directFromSellerSignalsPrefix|'s [=url/query=] is not null.
-      1. [=Assert=]: |directFromSellerSignalsPrefix|'s [=url/scheme=] is "`https`".
+      1. If |result| is not undefined or null, and typeof |result| is [=string=]:
+        1. Set |auctionConfig|'s [=auction config/direct from seller signals header ad slot=] to |result|.
+        1. Set |auctionConfig|'s [=auction config/expects direct from seller signals header ad slot=] to true.
     * To handle an error:
-      1. TODO: set the rep in |auctionConfig| to failure.
+      1. Set |auctionConfig|'s [=auction config/direct from seller signals header ad slot=] to failure.
+      1. Set |auctionConfig|'s [=auction config/expects direct from seller signals header ad slot=] to false.

```

Comment:
seems we don't need this boolean. Just check whether [=auction config/direct from seller signals header ad slot=] is failure would be the same as checking this variable being true/false.